recon/settings.go

```diff
@@ -94,20 +95,28 @@ func newIPMatcher() *ipMatcher {
 }
 
 func (m *ipMatcher) allow(partner Partner) error {
-	var httpAddr *net.TCPAddr
-	if partner.HTTPNet == NetworkDefault || partner.HTTPNet == NetworkTCP {
-		httpAddr, resolveErr := net.ResolveTCPAddr("tcp", partner.HTTPAddr)
-		if resolveErr == nil && httpAddr.IP != nil {
-			err := m.allowCIDR(fmt.Sprintf("%s/32", httpAddr.IP.String()))
-			if err != nil {
-				return errgo.Mask(err)
+	httpAddrs, resolveErr := resolveHost(partner.HTTPNet.String(), partner.HTTPAddr)
+	if resolveErr == nil && len(httpAddrs) > 0 {
+		for _, ip := range httpAddrs {
+			cidr := net.IPNet{IP: ip.IP}
```

Comment:
Perhaps pull this logic out into a function, reuse below? It'd also be more testable.